### DISCUSSION TITLE
Future of Fondo

### DISCUSSION THREAD

**Discussion by calo001** [calo001/fondo#145] [General]
Hi everyone!

The team of the Unsplash API has recently changed their API Guidelines. Point 3 of the guidelines indicates "You cannot replicate the core user experience of Unsplash (unofficial clients, wallpaper applications, etc.)". The way Fondo currently works violates the usage guideline [Guideline: Replicating Unsplash](https://help.unsplash.com/en/articles/2511257-guideline-replicating-unsplash).

Therefore, in order to not violating the Unsplash API Guidelines, Fondo needs to renew its main feature to become an app that even without the integration of the Unsplash API provides content and value to users.

In resume, Fondo must meet the following requirements:
* It can not be a wallpaper app that without the integration the app has no content and no value to users.
* It can not be an unofficial Unsplash client.

I'm looking for ideas and options to implement a suitable solution for this project, but at the same time, figure a way to keep the purpose of Fondo for which it was created "Find the most beautiful wallpapers".

**Comment by calo001** [comment]
My first idea is to create my own dedicated API where users can share their favorite wallpaper (photos, illustrations, vector images, etc). Doing this implies:
* Not to use the Unsplash API.
* Create a dedicated API for the Fondo apps (Gtk and Android clients)
* Research and evaluate alternatives for services to host the API and store files.

**Reply by vfrico** [reply]
To me, creating a custom API seems to be the only way. But it comes with the difficulty of creating a good community of people that  is willing to upload its own content to it.

Alternative services like pexels also discourages the usage of its images for this purposes.

**Reply by calo001** [reply]
II already did a search for other similar services, unfortunately, all the services I found have the same restriction.

In case of creating an API, this website wallpapercave.com could be a reference because of its focus on wallpapers. But as you say, the community will be a challenge.

**Reply by cassidyjames** [reply]
I would highly caution against rolling your own API/community that accepts user submissions; then you're suddenly on the hook if people share porn, copyrighted images, or worse.

**Reply by calo001** [reply]
That is just one of the tricky aspects that scares me even if I validate every file that is uploaded.

**Comment by cassidyjames** [comment]
Youch, this really sucks—I wonder if aggregating multiple sources (like Unsplash, Pexels, Pixabay, and maybe a Creative Commons Flickr filter or something?) would be acceptable to Unsplash, because then Unsplash itself is not providing 100% of the value?

**Reply by calo001** [reply]
Unfortunately, I think it is not valid, I quote you the answer I have from the Unsplash team: 

"wallpaper apps in general fall into the category of replicating the core user experience of Unsplash even if you are integrating other free platforms. Because your app is a wallpaper app, it currently violates the above API Usage Guideline."

**Comment by calo001** [comment]
I just found this API by https://wall.alphacoders.com/api.php, basically, the requirement to use their API are next:

Authentication

API authentication is achieved via an authentication key, every member can obtain a key via the following process.

    Register an Alpha Coders account
    Send an email to admin@alphacoders.com with the following information:
        Your Alpha Coders Username. Please send the email from the email you are registered on Alpha Coders with.
        Your Name/Contact Info
        Description of your project - including any public links.
        Example of your 'Powered by Wallpaper Abyss' branding (can be a work in progress).
        Whether you expect your site/app could eventually go over the free tier (150,000 queries/month).

Notes: For item #4, the 'Powered By Wallpaper Abyss' branding, we realize that different API implementations may find this difficult. Some examples are below:

    If developing a mobile app or website, the normal branding option shouldn't be a problem
    For bots used in Slack/Discord/etc If the image pulled links back to Wallpaper Abyss when clicked on, that would be OK as well (please send us a small code snippet)
    For something like a desktop wallpaper changer, we would require the branding to be placed where users find information to download your application (this could be a public github repo page), and/or present when someone opened the application.

**Comment by Suzie97** [comment]
How about showing wallpapers that will look good in elementary OS from Unsplash instead of the entire Unsplash collection? That wouldn't break their guidelines right?
